src/libos/src/fs/timer_file.rs

```diff
@@ -0,0 +1,193 @@
+use super::*;
+
+use crate::time::{itimerspec_t, timespec_t};
+use atomic::{Atomic, Ordering};
+use std::time::Duration;
+
+/// Native Linux timerfd
+#[derive(Debug)]
+pub struct TimerFile {
+    host_fd: HostFd,
+    host_events: Atomic<IoEvents>,
+    notifier: IoNotifier,
+}
+
+impl TimerFile {
+    pub fn new(clockid: u32, flags: TimerCreationFlags) -> Result<Self> {
+        let raw_host_fd = try_libc!({
+            let mut ret: i32 = 0;
+            let status = occlum_ocall_timerfd_create(&mut ret, clockid, flags.bits());
+            assert!(status == sgx_status_t::SGX_SUCCESS);
+            ret
+        }) as FileDesc;
+        let host_fd = HostFd::new(raw_host_fd);
+        let host_events = Atomic::new(IoEvents::empty());
+        let notifier = IoNotifier::new();
+        Ok(Self {
+            host_fd,
+            host_events,
+            notifier,
+        })
+    }
+
+    pub fn host_fd(&self) -> c_int {
+        self.host_fd.to_raw() as c_int
+    }
+
+    pub fn timerfd_settime(
```

Comment:
Too much unsafe here, I think you just wrap the `occlum_ocall_timerfd_settime` is enough.

this function can just return the timer value directly:
```
pub fn set_time(&self, new_value: &itimerspec_t) -> Result<itimerspec_t>
```

Reply:
Done